spec/abci/abci++_methods.md

```diff
@@ -444,6 +443,7 @@ the consensus algorithm will use it as proposal and will not call `RequestPrepar
       inform the Application of the block header's hash, which cannot be done at `PrepareProposal`
       time. In this case, the call to `RequestProcessProposal` occurs right after the call to
       `RequestPrepareProposal`.
+      However, there is no guarantee that the values of `RequestProcessProposal.txs` and `RequestProcessProposal.hash` will be the same as returned in the preceding `RequestPrepareProposal`.
```

Comment:
```
      However, there is no guarantee that the values of `RequestProcessProposal.txs` and `RequestProcessProposal.hash` will be the same as returned in the preceding call to `PrepareProposal`.
```suggestion

Also, we may need to clarify that `RequestProcessProposal.hash` is not returned by `RequestPrepareProposal`, but calculated by CometBFT just after `PrepareProposal` returns.